### PR TITLE
feat: data preparation

### PR DESCRIPTION
**Pull request by robincourant**
- Add a method to load a dataset
- Add a method to prepare data according to its type
- Add a CLI to launch these methods
- Create and update requirements
- Create gitignore
- Modify the README, by adding dataset descriptions

### CODE REVIEW

File: data_preparation.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return column_series
+
+
+def prepare_data(dataset_df: pd.DataFrame):
+    """Fill missing values and standardize float columns.
+
+    :param dataset_df: dataset to process.
+    """
+    for column_name, column_series in dataset_df.iteritems():
+        # Check whether the column type is boolean or integer
```
Comment (iqdecay) [inline data_preparation.py]: The comment doesn't add anything

File: data_preparation.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if is_integer_dtype(column_series):
+            if set(column_series.unique()) == {0, 1}:
+                dataset_df[column_name] = _prepare_bool(column_series)
+            else:
+                dataset_df[column_name] = _prepare_int(column_series)
+
```
Comment (iqdecay) [inline data_preparation.py]: Remove newline

**Review by iqdecay — CHANGES_REQUESTED** [review]
A few problems that are not obvious on review but come up in usage

File: data_preparation.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    :return: the processed column.
+    """
+    # Fill missing values with the mean or the median of the column
+    filling_value = column_series.mean() if mean else column_series.median()
+    column_series.fillna(int(filling_value), inplace=True)
+
+    return column_series
+
+
+def _prepare_bool(column_series: pd.Series):
```
Comment (iqdecay) [inline data_preparation.py]: Return type should be None (to inform that it is not supposed to return anything)

File: README.md [hunk 4]
```diff
@@ -12,3 +12,51 @@
...
+  - `INDUS`: proportion of non-retail business acres per town.
+  - `CHAS`: Charles River dummy variable (1 if tract bounds river; 0 otherwise).
+  - `NOX`: nitric oxides concentration (parts per 10 million).
+  - `RM`: average number of rooms per dwelling.
+  - `AGE`: proportion of owner-occupied units built prior to 1940.
+  - `DIS`: weighted distances to five Boston employment centres.
+  - `RAD`: index of accessibility to radial highways.
+  - `TAX`: full-value property-tax rate per $10,000.
+  - `PTRATIO`: pupil-teacher ratio by town.
+  - `B`: 1000(Bk - 0.63)^2 where Bk is the proportion of blacks by town.
```
Comment (iqdecay) [inline README.md]: wording ???

File: README.md [hunk 5]
```diff
@@ -12,3 +12,51 @@
...
+python main.py dataset_filename
+```
+
+## Datasets
+
+### Boston Housing:
+
+Download this dataset [here](https://www.kaggle.com/altavish/boston-housing-dataset).
+
+There are 13 features to predict the price houses in Boston.
```
Comment (iqdecay) [inline README.md]: What column is the target  ?

File: data_preparation.py (hunk 3 above)
Comment (iqdecay) [inline data_preparation.py]: EDIT : comment in the wrong place

File: data_preparation.py (hunk 1 above)
Comment (robincourant) [reply data_preparation.py]: Removed

File: data_preparation.py (hunk 2 above)
Comment (robincourant) [reply data_preparation.py]: Done

File: data_preparation.py (hunk 3 above)
Comment (robincourant) [reply data_preparation.py]: Ups ! Worth than that, the method returns a series ... Fixed

File: README.md (hunk 5 above)
Comment (robincourant) [reply README.md]: Added

File: README.md (hunk 4 above)
Comment (robincourant) [reply README.md]: Ok PC guy

File: data_preparation.py (hunk 3 above)
Comment (robincourant) [reply data_preparation.py]: Removed the wrong comment below. There is nothing to add for me, we just replace missing values by the most frequent value.

File: README.md (hunk 4 above)
Comment (iqdecay) [reply README.md]: <3

**Review by iqdecay — CHANGES_REQUESTED** [review]
Still need to add the various options for data preparation in the prepare_data

File: data_preparation.py (hunk 3 above)
Comment (iqdecay) [reply data_preparation.py]: EDIT : comment in the wrong place

File: data_preparation.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    """
+    # Fill missing values with the mean or the median of the column
+    filling_value = column_series.mode()
+    column_series.fillna(int(filling_value), inplace=True)
+
+    return column_series
+
+
+def prepare_data(dataset_df: pd.DataFrame):
+    """Fill missing values and standardize float columns.
```
Comment (iqdecay) [inline data_preparation.py:81]: There should be parameters on how to prepare the data : drop na, replace by mean, value. Here it normalizes everything
Return type should be None

**Review by iqdecay — CHANGES_REQUESTED** [review]
Add parameters for data preparation

**Pull request merged**